alertmanager/peers.go

```diff
@@ -0,0 +1,87 @@
+package alertmanager
+
+import (
+	"net"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/log"
+)
+
+var (
+	srvRequestDuration = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace: "cortex",
+		Name:      "srv_lookup_request_duration_seconds",
+		Help:      "Time spent looking up SRV records.",
+		Buckets:   prometheus.DefBuckets,
+	}, []string{"service", "proto", "hostname", "status_code"})
+)
+
+func init() {
+	prometheus.MustRegister(srvRequestDuration)
+}
+
+// TODO: change memcache_client to use this.
+
+// SRVDiscovery discovers SRV services.
+type SRVDiscovery struct {
```

Comment:
I want to use it for memcached. I guess I'll make it public when that actually happens.